hashicorp/consul-k8s
# Fix "consul-k8s install fails: now can't re-install or upgrade"

Fixes #1005 

Changes proposed in this PR:
- Testing
  - Extend `HelmCluster` and `CLICluster` to accept arguments on `Create`, `Update`, and `Delete` actions.
  - Allow tests to set the namespace used for CLI acceptance tests.
  - Add a scenario test to exercise upgrading Consul after a failed install.
  - Add a scenario test to exercise upgrading Consul installed in the `default` namespace.
- Bug fix
  - Fix an issue in the upgrade case where the cluster was being found by namespace instead of its name. 

How I've tested this PR:

Both using scenario tests and by debugging the upgrade command when an install was present in the `default` namespace.

How I expect reviewers to test this PR:

The scenario tests may be run from the `acceptance` directory with

``` bash
go test -timeout 15m github.com/hashicorp/consul-k8s/acceptance/tests/cli -v
```

Checklist:
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)
